pointer-masking-proposal.adoc

```diff
@@ -158,41 +155,47 @@ WPRI,UTE,WPRI,PM
 Figure 3d: Memory Tagging Extension register (_**umte**_) for RV128 (U-mode) +
  +
 
-[cols=",",]
+[cols="15%,15%,70%", options="header"]
 |===========================================================================================================================
-|*Value* |*Meaning*
-|00 |PM is disabled (_default_) and _**xPMMASK**_ and _**xPMBASE**_ registers can only be modified by the higher privilege level
-|01 |PM is enabled and _**xPMMASK**_ and _**xPMBASE**_ registers can only be modified by the higher privilege level
-|10 |PM is disabled and _**xPMMASK**_ and _**xPMBASE**_ registers can be modified by the same privilege level
-|11 |PM is enabled and _**xPMMASK**_ and _**xPMBASE**_ registers can be modified by the same privilege level
+^|*PM Bits* ^|*Name* ^|*Meaning*
+^.^| PM[0:0] ^.^| PM.Enabled |
+
+  0 – PM is disabled (_default_) +
+  1 – PM is enabled
+
+^.^| PM[1:1] ^.^| PM.Current |
+
+ 0 – _**xPMMASK**_ and _**xPMBASE**_ registers can only be modified by the higher privilege level +
```

Comment:
With PM.Current 2-3 registers have unique way to check privilege level access: the usual one with bits [9:8] from the CSR address and PM.Current bit. This introduces extra complexity in terms of accessing CSR. Do you think it's worth it?

Reply:
The purpose of PM.Current not about checking priv level access but is to be able to "assign" the rights of enabling / disabled PM enforcement to the current privilege level (e.g. modify bit PM.Enable). By default, only higher privilege level has the rights to disable (and enable) PM enforcement for lower (current) privilege levels. However, during the discussion on the J-ext group we come up to the conclusion that from the U-mode runtime environment perspective it might be desired to allow an option to modify it from the current privilege level.
Another use case for a self-manage PM enforcement is an in-process isolation feature.